### PR TITLE
added login controller & route

### PR DESCRIPTION
**Pull request by criticic**
Issue: #4

#### Short description of what this resolves:
Added Login Controller, which returns a 200 response with message "User Logged in Successfully"
Linked the controller with the route

### CODE REVIEW

File: src/controllers/userAuth.controllers.js [hunk 1]
```diff
@@ -45,5 +45,43 @@ async function registerUser(req, res) {
...
 }
 
-export default registerUser;
+async function loginUser(req, res) {
+    try {
+        const { email, password } = req.body;
+
+        const user = await prisma.user.findUnique({
+            where: {
+                email: email
```
Comment (bhupesh98) [inline src/controllers/userAuth.controllers.js:54]: filter out the data required, this make SQL query a bit efficient.

File: src/controllers/userAuth.controllers.js [hunk 2]
```diff
@@ -45,5 +45,43 @@ async function registerUser(req, res) {
...
+        if (!isMatch) {
+            console.log('Error logging in: Incorrect password');
+            res.status(401).json({
+                error: 'Incorrect password'
+            });
+            return;
+        }
+
+        res.status(200).json({
+            message: 'User logged in successfully'
```
Comment (bhupesh98) [inline src/controllers/userAuth.controllers.js]: If user is found then pass out some relevant data.

**Review by bhupesh98 — APPROVED** [review]

File: src/controllers/userAuth.controllers.js (hunk 1 above)
Comment (criticic) [reply src/controllers/userAuth.controllers.js:54]: now I am returning all of the user's data on login, so is it sill needed?

Comment (bhupesh98) [reply src/controllers/userAuth.controllers.js:54]: Heyy @criticic  there are some unrequired data like roomId which is used for referencing Room, which is typically not required during login and hashedPassword need not to be shared. Refer to prisma docs and look for select statements for filtering out data.

Comment (criticic) [reply src/controllers/userAuth.controllers.js:54]: @bhupesh98 kindly take a look now

File: src/controllers/userAuth.controllers.js [hunk 3]
```diff
@@ -52,6 +52,13 @@ async function loginUser(req, res) {
         const user = await prisma.user.findUnique({
             where: {
                 email: email
+            },
+            select: {
+                id: true,
+                name: true,
+                email: true,
+                password: true,
```
Comment (VatsalBhuva11) [inline src/controllers/userAuth.controllers.js]: no need to select the password field. remove the select for password, and directly return the obtained user object below

Comment (criticic) [reply src/controllers/userAuth.controllers.js]: @VatsalBhuva11 I have added it

**Pull request merged**